_drafts/2022-03-12-load-fast.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: post
+title: "Load FAST Pharo/Java model"
+subtitle: How to generate the AST of your model using FAST !
+header-img: img/posts/2022-03-14-load-fast/bg-post.jpg
+author: Ahmed Zaki BENNECER
+background: "/img/posts/2022-03-14-load-fast/bg-post.jpg"
+date: 2022-03-10 10:00:00
+comments: true
+---
+
+# Introduction
+
+When we are interested in the migration/modernization of projects we talk about modeling or meta models. Moose revolves around a powerful Famix meta model that allows us to do several operations. For instance, previous posts present [the analysis and requesting a model](https://modularmoose.org/2021/03/30/modularmoose-usecase.html), [visualizing a model with plantUML](https://modularmoose.org/2021/06/04/plantUML-for-metamodel.html) or even [creating a model](https://modularmoose.org/2021/02/15/Coasters.html), etc.
+
+FAST is a meta-model that helps us understand source code in a less abstract way. Indeed, FAST is based on AST (Abstract Syntax Tree) which is close to the source code. And as the devil is in the details, FAST contains interesting elements when analyzing programs (for example some specific expression statement), and effectively this is what makes the difference between FAST and FAMIX.
+(Consult this [overview](https://modularmoose.org/moose-wiki/Developers/Parsers/FAST) about the FAST model).
+
+![Abstraction Level](/img/posts/2022-03-14-load-fast/abstraction.png)
+
+In this blog post we will talk about how to load, generate a FAST model of Java code. For this we will take ArgoUML, an open-source Java project, as an example.
+
+# Overview & Prerequisites
+
+First of all, we have to understand from where we are going to start and where we are going to end up. As already mentioned, we will take the AgroUML’s java code and the goal is to generate the corresponding AST and do analyses above. To do this, 3 steps are necessary to have the AST as illustrated in the figure below:
+
+1. Parse Java to build FAMIX model
+2. Load the model into Moose
+3. Generate the AST
+
+![Steps for generating FAST Model](/img/posts/2022-03-14-load-fast/Overview.png)
+
+Before starting, it is better to download [the source code](https://www.mediafire.com/file/052scdvr07ycok8/ArgoUML-0-34.zip/file) and [the famix model](https://drive.google.com/file/d/1ZQQYedGKWlC7whj92uErpfbI7T-kSjGC/view) of the AgroUML project, step 1 of the diagram above (follow this [blog](https://fuhrmanator.github.io/2019/07/29/AnalyzingJavaWithMoose.html) for more details).
```

Comment:
Please upload the ressources in this git repository instead of using external link